### ISSUE TITLE
Request to become a collaborator

### ISSUE THREAD

**Issue by sorin-ionescu** [sorin-ionescu/prezto#1269]
If thou lovest Prezto and wishest thyself a collaborator (maintainer), post thy username here and a short description of thyself.

**Comment by belak** [comment]
Username: belak

About me: My full name is Kaleb Elwert (as you can see in my commits). I'm primarily a go and python developer (hence my interest in the python/virtualenv portions) but have a bit of experience with random shell scripting as well.

What I think could be improved: Small fixes for current packages and maintaining compatibility would be the focus. Any additions in the future would have to be shown to be useful to more than one person before being considered and configuration for existing themes/packages would be preferred over even more modules.

Why I'm interested: prezto has been a part of my dotfiles for quite a while and it would be great to contribute back to it. My frustrations lately have been because both of the major configuration frameworks (prezto and OMZ, though I prefer prezto for the simplicity) seem to get sparse updates so I'd love to be able to help with that.

**Comment by sorin-ionescu** [comment]
I have sent collaboration invites to @facastagnini, @johnpneumann, @paulmelnikow, and @belak.

**Comment by facastagnini** [comment]
Hi @sorin-ionescu! Man I am sorry for going silent since you came back, but I am really busy at the moment and I dont have much free time. Thanks for the collaboration invite, before accepting, is that ok if I give some love to prezto sporadically? At this moment I cant commit to be hands on every day :/

**Comment by sorin-ionescu** [comment]
@facastagnini, I don't expect collaborators to work on it daily. It's fine.

**Comment by belak** [comment]
Do you have any specific expectations/limitations for collaborators? This has been discussed a little bit elsewhere but it would be nice to have it here so there's no confusion.

**Comment by sorin-ionescu** [comment]
I will not limit collaborators as long as they remain true to the project by not turning into something it is not.

I expect collaborators to continue the work in the same manner it has been done, for it is the reason why we are all here.

A collaborator should mind the coding and comment style, write good commit messages in correct English per the documentation, and not create a commit log graph that would make an autistic savant with a doctorate in graph theory throw his laptop out the window; thus, I prefer to rebase even when dates become out of order.

On occasion, to not annoy pull-request submitters with pendantics, I modify their patches and sign them.

**Comment by maximbaz** [comment]
@belak thanks for starting to merge pull requests! 

Just a question, there have been a few improvements and bug fixes merged in [zsh-users/prezto](https://github.com/zsh-users/prezto), are you planning to backport them here, or the authors have to recreate the pull requests for this repo?

**Comment by belak** [comment]
@maximbaz I'd much prefer to have the authors recreate them here. I'm doing my best to go through existing pull requests against this repo and get them merged in, so trying to manage side-porting pull requests from another repo would get way too complicated for me.

@sorin-ionescu Is there an official stance on new modules and/or new themes that don't have a ton of people interested in them? I've been holding off on those PRs and focusing on bug-fixes and improvements for the time being because I want to know what your opinions are on them. I'd like to avoid having 50 open PRs for adding new modules if possible... and whether that's through politely declining or merging them, I don't have a huge preference.

**Comment by jeffwidman** [comment]
I'm interested in helping maintain prezto. 

I pitch in on an as-time-allows basis on a number of projects, particularly in the python world. I'm an active collaborator on Flask, FactoryBoy, kafka-python, etc. 

I don't have one single thing that I want for the project other than to see it stay active and lean. When in doubt saying "no" rather than "yes" to feature requests, as I think oh-my-zsh does a great job of filling the void for those who want everything. 

For most of the projects I'm involved with, I don't write a lot of new code, more I review PR's and help triage the issue tracker.

**Comment by sorin-ionescu** [comment]
@belak Themes are personal, tend to be used by the author, nobody else, and tend to be slight modifications of existing themes; thus, I often reject them. Popular themes should be merged since users will look for them. Beautiful, creative themes should also be merged. Obscure modules should not be merged. For example, we'll have trouble maintaining modules for domain specific software none of us use. Prezto must remain lean.

@belak, @maximbaz Let's keep issues on topic. This issue is about requesting to become a contributor. Thank you.

**Comment by sorin-ionescu** [comment]
Invitation sent to @jeffwidman.

**Comment by jeffwidman** [comment]
> Invitation sent to @jeffwidman.

Thanks, I will try to start chipping in soon.

**Comment by indrajitr** [comment]
I continue to use prezto over last few years. And would be very much interested in helping.

There are some personal patches and updates that I'd like to push upstream (via PR). Additionally, would be happy to review other PRs (and possibly triage the exiting ones).

**Comment by sorin-ionescu** [comment]
@indrajitr I remember you. You contributed to zsh-history-substring-search. Invitation sent.

**Comment by indrajitr** [comment]
> @indrajitr I remember you. You contributed to zsh-history-substring-search. Invitation sent.

Appreciated, thank you!

**Comment by sorin-ionescu** [comment]
@belak, @indrajitr, you have done a wonderful job. Thank you.

**Comment by belak** [comment]
Thanks! I haven't been as active lately, but it's nice to hear that you don't think we've let it fall apart in your absence. :)

**Comment by sorin-ionescu** [comment]
Shell wizard lovers of Prezto, please apply to get write access. Thank you.

**Comment by belak** [comment]
One person who comes to mind is @romkatv because of the updates to Powerlevel10k. I've been merging those updates for releases but I can't always keep up with those.

**Comment by romkatv** [comment]
@belak I'm honored but cannot accept the nomination. I have only rudimentary familiarity with prezto and don't use it myself. My general zsh experience is also lacking as I've started using it only half a year ago.

**Comment by srijanshetty** [comment]
While I don't have a list of open source projects that I've contributed to (employment constraints), I've been using Prezto for half a decade now and would like to contribute in whatever way possible to keep this project alive.

**Comment by JARVIS-AI** [comment]
I would like to collaborate to make this project alive 
Even there are contributions already made for prezto
I would like to join other developers for prezto

**Comment by sorin-ionescu** [comment]
@srijanshetty, you’ve been invited.

**Comment by sorin-ionescu** [comment]
@JARVIS-AI, thank you; however, you’re an anonymous account with an Iranian address. Where do you fall on the U.S. Government’s [naughty list](https://www.treasury.gov/resource-center/sanctions/Programs/Pages/iran.aspx)? Parts of GitHub are [restricted](https://www.theverge.com/2019/7/29/8934694/github-us-trade-sanctions-developers-restricted-crimea-cuba-iran-north-korea-syria) to you.

**Comment by srijanshetty** [comment]
@sorin-ionescu: Thanks a lot for the invite.

**Comment by wadkar** [comment]
@sorin-ionescu long time prezto fan, would be happy to help keeping this project alive and maintained!

**Comment by Shaunakde** [comment]
Dear Sorin, 

I'll start this out with a little story. I'm a long time bash user, forced by Apple's newest update to switch to zsh (or endure warnings in the terminal for all eternity).

I just. That's not completely true. I have been wanting to switch to zsh for years, but I've been misusing bashisms for a while, and I feared to switch would be uncomfortable. When I updated my OS I decided this was a sign to make the switch. I got myself a copy of [moving to zsh](https://scriptingosx.com/2019/06/moving-to-zsh/) and dived right in. 

Off the bat, I had to replace my PS string and a few google searches later I landed on OMZ. As one does, I found my way to the original PR that lead to the creation of this project and read through the whole exchange. On the way, I was convinced by many of the philosophical directions that were indicated to have led to the creation of this repository, and, here I am. 

As I come up to speed with my zsh knowledge, I'd love to help fix bugs, documentation and test issues for prezto!

About me: 
I describe myself as a scientific computation person. I used to write numerical routines in C, and design and develop algorithms to let computers understand signals. These days I do the same in python. I live and work in San Franciso, and I enjoy walking around the city sketching. 

Shaunak

**Comment by romkatv** [comment]
@Shaunakde Can you share some of your sketches?

**Comment by Shaunakde** [comment]
Very happy to oblige

![Sketch_Prezto](https://user-images.githubusercontent.com/2545129/78815432-1fd6bf00-7985-11ea-9546-26835e554657.png)

**Comment by romkatv** [comment]
298x273! I appreciate the minimalist approach you are taking. You should fit right in.

**Comment by belak** [comment]
@Shaunakde Thanks for commenting! We're always looking for new contributors! There are a lot of issues still open we haven't had a chance to get to yet, so if you've got the time, submitting PRs is always super helpful.

A few things to note: we haven't been accepting module changes here (because we haven't completely agreed on what the bar is) but there's a [prezto-contrib](https://github.com/belak/prezto-contrib) repo for new modules/themes.

**Comment by jeffwidman** [comment]
In general, if you want to become a maintainer, then prove you're around for the long haul by showing up over a couple of months helping answer questions on issues or submitting patches. 

Not hard to do, and in fact a little secret is that often the most powerful work a collaborator does isn't fixing all the bugs but rather reviewing all the code and tracking what happened so that drive-by patches can be applied in a consistent manner and leverage existing features elsewhere in the codebase that that someone may not realize unless they've been around for a while. That and triaging issues from n00bs. So code productivity matters less than the willingness to stick around for a time... I've found that across many projects, folks often lose interest quickly... but if you're willing to stick around and aware of when to ask for help rather than blindly approving patches, then we are very happy to have someone join as a collaborator.

**Comment by jeffwidman** [comment]
Along those lines I support making @wadkar a collaborator... he's consistently been involved in this project. cc @belak @johnpneumann @sorin-ionescu

**Comment by Shaunakde** [comment]
@jeffwidman Absolutely agree. Will hang around, track issues and evaluate them the best I can :)

**Comment by jeffwidman** [comment]
I also support @hotpxl becoming a maintainer if he's interested... he's been involved for years: https://github.com/sorin-ionescu/prezto/pulls?q=is%3Apr+author%3Ahotpxl+

cc @belak @indrajitr @johnpneumann @sorin-ionescu

**Comment by belak** [comment]
I'm on board with that.

I would also strongly encourage [moving this repo to an org like zsh-users](https://github.com/sorin-ionescu/prezto/issues/1284). This would make it so adding a new maintainer can be done faster in the future, and avoid the need to wait on Sorin.

Honestly, this is one of the reasons why I've been around less: I lost interest in Prezto. When the primary maintainer is MIA, there's no way to add other collaborators without them, and nobody seems to have the authority to make larger changes/improvements, it ends up being very draining to work on that project, especially when it's not clear if it will change in the future.

**Comment by huyz** [comment]
Those are good ideas.  What do you think @sorin-ionescu ?
We love what you've done and created.  We don't want this project to languish and disappear.  Let's save your baby!

**Comment by hotpxl** [comment]
@jeffwidman I really appreciate the nomination! Thank you all for the support
